### DISCUSSION TITLE
No logging or error output when plugin installation fails.

### DISCUSSION THREAD

**Discussion by henrypenny** [craftcms/cms#8804] [Ideas]
### Description

If there's an error during a plugin install migration. If fails silently with no exception trace in the browser or logs. Just: 'The plugin was not installed'

        craft\base\Plugin.php line 138:

        // Run the install migration, if there is one
        if (($migration = $this->createInstallMigration()) !== null) {
            try {
                $migrator->migrateUp($migration);
            } catch (MigrationException $e) {
                return false;
            }
        }

Spent an hour in xdebug to find out that the table already existed.

IMO it would be better to spill the exception to the browser if you are running in dev mode.
And it should be logged in all circumstances, otherwise you're completely blind to why the plugin wouldn't install.

Migration Error – craft\errors\MigrationException
An error occurred while executing the "dolphiq\redirect\migrations\Install migration: SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'dolphiq_redirects' already exists
The SQL being executed was: CREATE TABLE `dolphiq_redirects` (
`id` int(11) NOT NULL AUTO_INCREMENT PRIMARY KEY,
`sourceUrl` varchar(255),
`destinationUrl` varchar(255),
`statusCode` varchar(255),
`hitCount` int(11) UNSIGNED NOT NULL DEFAULT 0,
`hitAt` datetime(0),
`dateCreated` datetime(0) NOT NULL,
`dateUpdated` datetime(0) NOT NULL,
`uid` char(36) NOT NULL DEFAULT '0'
) ENGINE=InnoDb DEFAULT CHARSET=utf8

**Comment by brandonkelly** [comment]
The error should get logged to `storage/logs/web.log` – did you check there?

**Comment by henrypenny** [comment]
No I didn't. The code doesn't appear to do that. Is this behind the scenes somehow?

**Comment by henrypenny** [comment]
Also that file fills up with massive amounts of information. Isolating a specific issue is very difficult.

`cat storage/log/web.log | grep <what>?`

If you don't know why the install failed, what to you search for?

**Comment by brandonkelly** [comment]
Yeah, migration exceptions will be printed out, as migrations are often run in the terminal:

https://github.com/craftcms/cms/blob/2613dc758987a2d986ccf0a0c4f59b4c303eef1c/src/db/Migration.php#L57-L58

https://github.com/craftcms/cms/blob/2613dc758987a2d986ccf0a0c4f59b4c303eef1c/src/db/Migration.php#L429-L434

If the migration is not being run in the terminal, then their output will be caught by an output buffer, and if an error occurred, it will be logged as an error to `storage/logs/web.log`:

https://github.com/craftcms/cms/blob/2613dc758987a2d986ccf0a0c4f59b4c303eef1c/src/db/MigrationManager.php#L243-L249

**Comment by henrypenny** [comment]
I would still like exceptions sent to the browser in any case. As I said I burned an hour finding out something very simple.
